lib/functions/plot_measures.R

```diff
@@ -94,12 +131,62 @@ plot_measures <- function(
     plot_tmp <- plot_tmp +
       facet_wrap(vars({{ facet_var }}), ncol = 2)
   }
+  # Add y_scale to add option for free_y
+  if (!is.null(y_scale) && y_scale == "free_y") {
+    plot_tmp <- plot_tmp +
+      facet_wrap(~source, scales = "free_y")
+  }
+
+  # Add save_path option
+  if (!is.null(save_path)) {
+    ggsave(
+      filename = here("released_output", "results", "figures", save_path),
+      plot = plot_tmp,
+      width = 10,
+      height = 6
+    )
+  }
 
   plot_tmp
 }
 
+# Combining two figures into one using patchwork
+patch_figures <- function(figure_1, figure_2, save_path = NULL) {
+  combined_figure <- (figure_1 + figure_2) +
+    plot_annotation(tag_levels = "A") +
+    plot_layout(guides = "collect", widths = c(2, 1)) &
+    theme(
+      legend.position = "bottom",
+      text = element_text(size = 15),
+      strip.background = element_rect(size = 0),
+      strip.text.x = element_text(size = 13, face = "bold")
+    )
+
+  if (!is.null(save_path)) {
+    ggsave(
+      filename = here("released_output", "results", "figures", save_path),
+      plot = combined_figure,
+      width = 15,
+      height = 6
+    )
+  }
+  combined_figure
+}
```

Comment:
as mentioned above, I think this should be a separate call to keep the functions smallish